src/main/java/com/cloudant/client/org/lightcouch/CouchDbClientBase.java

```diff
@@ -512,17 +479,114 @@ public Gson getGson() {
     /**
      * Executes a HTTP request and return a domain object
      *
-     * @param request  The HTTP request to execute.
-     * @param Class<T> class of domain object to return
-     * @return <T> T
+     * @param connection  The HTTP connection to execute.
+     * @return <T> T json object
      */
-    public <T> T executeRequest(HttpRequestBase request, Class<T> classType) {
-        HttpResponse response = null;
+    public InputStream executeRequest(HttpConnection connection) {

```

Comment:
Why do we need this method? Can't the `executeToInputStream` on line 495 just be public?